atom/browser/api/atom_api_session.cc

```diff
@@ -330,6 +341,85 @@ void OnClearStorageDataDone(const base::Closure& callback) {
     callback.Run();
 }
 
+void DidReadBlobData(const scoped_refptr<net::IOBuffer>& blob_data,
+                     const Session::BlobDataCallback& completion_callback,
+                     int bytes_read) {
+  std::unique_ptr<base::BinaryValue> result(
```

Comment:
Converting `BinaryValue` to `Buffer` will trigger a copy of the data, which is unnecessary since the data is already copied here. A more efficient way is to use the copied data to construct `Buffer` directly.

An example can be found at `chromium_src/chrome/browser/printing/print_preview_message_handler.cc`.